src/main/java/org/eclipse/yasson/internal/serializer/MapSerializer.java

```diff
@@ -34,14 +35,21 @@ protected MapSerializer(SerializerBuilder builder) {
         super(builder);
     }
 
-    @SuppressWarnings("unchecked")
     @Override
     protected void serializeInternal(T obj, JsonGenerator generator, SerializationContext ctx) {
         for (Map.Entry<?,?> entry : obj.entrySet()) {
-            final String keysString = String.valueOf(entry.getKey());
+            final Object key = entry.getKey();
+            if (key == null) {
```

Comment:
It seems as though we should only be allowing null property values, but not null property keys.

Reply:
hi @aguibert, I found that for `null` keys serialization broken in either case:

```
    @Test
    public void testSerializeAndDeserializeMapWithNullKeys() {
        Jsonb jsonb = JsonbBuilder.create(new JsonbConfig().withNullValues(Boolean.TRUE));
        Map<Object, String> nullKey = singletonMap(null, "value");
        assertEquals(nullKey, jsonb.fromJson(jsonb.toJson(nullKey), Map.class));
    }
```

So I would ignore null values on both sides, and fix the issue with `null` keys separately